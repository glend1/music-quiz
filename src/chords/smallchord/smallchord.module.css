.chord {
	/* border: 1px solid var(--bg-colour); */
	border-radius: 0.3rem;
	margin: 1rem;
	/* padding: 1rem; */
	cursor: pointer;
}

.chord:hover {
	background-color: var(--bg-colour);
}

.chord img {
	float: right;
	margin: 0.3rem;
}

.chord div {
	margin: 0rem 1rem 0rem 1rem;
}

.chord span {
	display: block;
	margin: 0.5rem;
}

.chord.active {
	background-color: var(--dark-outline-colour);
}
