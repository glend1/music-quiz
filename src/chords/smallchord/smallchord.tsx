import { Dispatch, MouseEventHandler, SetStateAction } from "react";
import { Piano } from "../../canvas/piano/deletepiano";
import Image from "next/image";
import { cross } from "../../elements/images";
import { getSingleChord, IStdNote, StdNote } from "../../notes/notes/notes";
import { TChords } from "../types";
import styles from "./smallchord.module.css";

export function SmallChord({
	chords,
	active,
	setChords,
	id,
	setActive,
}: {
	active: string;
	setActive: Dispatch<SetStateAction<string>>;
	chords: TChords;
	setChords: Dispatch<SetStateAction<TChords>>;
	id: string;
}) {
	let clickEvent: MouseEventHandler<HTMLElement> = (e) => {
		setActive(id);
	};
	let selected: IStdNote[] = [];
	if (chords[id]) {
		selected = chords[id].map((i) => {
			return StdNote(i);
		});
	}
	let chord = getSingleChord(chords[id]);
	return (
		<div
			onClick={clickEvent}
			className={`${styles.chord} ${active == id ? styles.active : null}`}
		>
			<Image
				onClick={(e) => {
					e.stopPropagation();
					if (id == active) {
						let ids = Object.keys(chords);
						let index = ids.indexOf(id);
						let previousId = ids[index - 1];
						if (previousId) {
							setActive(previousId);
						}
					}
					setChords((prev) => {
						delete prev[id];
						return { ...prev };
					});
				}}
				height={25}
				src={cross}
				alt="Delete Chord"
			/>
			<span>{chord}</span>
			<Piano
				width={100}
				height={50}
				highlight={selected}
				cb={function (): void {}}
			/>
		</div>
	);
}
