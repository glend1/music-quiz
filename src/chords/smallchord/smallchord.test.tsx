import { render } from "@testing-library/react";
import { useState } from "react";
import { TChords } from "../types";
import { SmallChord } from "./smallchord";

function MockSmallChord() {
	const [active, setActive] = useState("test");
	const [chords, setChords] = useState<TChords>({ test: [] });
	return (
		<SmallChord
			active={active}
			setActive={setActive}
			chords={chords}
			setChords={setChords}
			id={active}
		/>
	);
}

describe("smallChord", () => {
	it("Should render a default component", () => {
		let component = render(<MockSmallChord />);
		expect(component.baseElement).toMatchSnapshot();
	});
});
