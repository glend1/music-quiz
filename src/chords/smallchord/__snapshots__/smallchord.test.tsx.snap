// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smallChord Should render a default component 1`] = `
<body>
  <div>
    <div
      class="chord active"
    >
      <img
        alt="Delete Chord"
        data-nimg="1"
        decoding="async"
        height="25"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
        style="color: transparent;"
        width="25"
      />
      <span />
      <div>
        <svg
          viewBox="0 0 100 50"
          width="100"
        >
          <g
            width="14.285714285714286"
          >
            <path
              class="whiteKey key "
              d="M 0 0 v 47.5 q 0 2.5 1.4285714285714288 2.5 h 11.428571428571429 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -47.5 Z"
              data-natural="C"
              data-octave="lower"
              rx="1.4285714285714288"
              ry="2.5"
              x="0"
            />
          </g>
          <g
            width="14.285714285714286"
          >
            <path
              class="whiteKey key "
              d="M 14.285714285714286 0 v 47.5 q 0 2.5 1.4285714285714288 2.5 h 11.428571428571429 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -47.5 Z"
              data-natural="D"
              data-octave="lower"
              rx="1.4285714285714288"
              ry="2.5"
              x="14.285714285714286"
            />
          </g>
          <g
            width="7.142857142857143"
          >
            <path
              class="blackKey key "
              d="M 10.714285714285715 0 v 28.75 q 0 2.5 1.4285714285714288 2.5 h 4.285714285714286 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -28.75 Z"
              data-flat="Db"
              data-octave="lower"
              data-sharp="C#"
              rx="1.4285714285714288"
              ry="2.5"
              x="10.714285714285715"
            />
          </g>
          <g
            width="14.285714285714286"
          >
            <path
              class="whiteKey key "
              d="M 28.571428571428573 0 v 47.5 q 0 2.5 1.4285714285714288 2.5 h 11.428571428571429 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -47.5 Z"
              data-natural="E"
              data-octave="lower"
              rx="1.4285714285714288"
              ry="2.5"
              x="28.571428571428573"
            />
          </g>
          <g
            width="7.142857142857143"
          >
            <path
              class="blackKey key "
              d="M 25 0 v 28.75 q 0 2.5 1.4285714285714288 2.5 h 4.285714285714286 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -28.75 Z"
              data-flat="Eb"
              data-octave="lower"
              data-sharp="D#"
              rx="1.4285714285714288"
              ry="2.5"
              x="25"
            />
          </g>
          <g
            width="14.285714285714286"
          >
            <path
              class="whiteKey key "
              d="M 42.85714285714286 0 v 47.5 q 0 2.5 1.4285714285714288 2.5 h 11.428571428571429 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -47.5 Z"
              data-natural="F"
              data-octave="lower"
              rx="1.4285714285714288"
              ry="2.5"
              x="42.85714285714286"
            />
          </g>
          <g
            width="14.285714285714286"
          >
            <path
              class="whiteKey key "
              d="M 57.142857142857146 0 v 47.5 q 0 2.5 1.4285714285714288 2.5 h 11.428571428571429 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -47.5 Z"
              data-natural="G"
              data-octave="lower"
              rx="1.4285714285714288"
              ry="2.5"
              x="57.142857142857146"
            />
          </g>
          <g
            width="7.142857142857143"
          >
            <path
              class="blackKey key "
              d="M 53.57142857142857 0 v 28.75 q 0 2.5 1.4285714285714288 2.5 h 4.285714285714286 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -28.75 Z"
              data-flat="Gb"
              data-octave="lower"
              data-sharp="F#"
              rx="1.4285714285714288"
              ry="2.5"
              x="53.57142857142857"
            />
          </g>
          <g
            width="14.285714285714286"
          >
            <path
              class="whiteKey key "
              d="M 71.42857142857143 0 v 47.5 q 0 2.5 1.4285714285714288 2.5 h 11.428571428571429 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -47.5 Z"
              data-natural="A"
              data-octave="lower"
              rx="1.4285714285714288"
              ry="2.5"
              x="71.42857142857143"
            />
          </g>
          <g
            width="7.142857142857143"
          >
            <path
              class="blackKey key "
              d="M 67.85714285714286 0 v 28.75 q 0 2.5 1.4285714285714288 2.5 h 4.285714285714286 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -28.75 Z"
              data-flat="Ab"
              data-octave="lower"
              data-sharp="G#"
              rx="1.4285714285714288"
              ry="2.5"
              x="67.85714285714286"
            />
          </g>
          <g
            width="14.285714285714286"
          >
            <path
              class="whiteKey key "
              d="M 85.71428571428572 0 v 47.5 q 0 2.5 1.4285714285714288 2.5 h 11.428571428571429 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -47.5 Z"
              data-natural="B"
              data-octave="lower"
              rx="1.4285714285714288"
              ry="2.5"
              x="85.71428571428572"
            />
          </g>
          <g
            width="7.142857142857143"
          >
            <path
              class="blackKey key "
              d="M 82.14285714285715 0 v 28.75 q 0 2.5 1.4285714285714288 2.5 h 4.285714285714286 q 1.4285714285714288 0 1.4285714285714288 -2.5 v -28.75 Z"
              data-flat="Bb"
              data-octave="lower"
              data-sharp="A#"
              rx="1.4285714285714288"
              ry="2.5"
              x="82.14285714285715"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</body>
`;
